druid-shell/src/windows/mod.rs

```diff
@@ -391,7 +391,7 @@ impl WndProc for MyWndProc {
                 if let Ok(mut s) = self.state.try_borrow_mut() {
                     let s = s.as_mut().unwrap();
                     if s.dcomp_state.is_some() {
-                        let mut rect: RECT = mem::uninitialized();
```

Comment:
This stuff is new to me too, but reading the docs for `MaybeUninit` it doesn't sound like `assume_init` is what we want. I think we want to just use `uninit()`, and then call `assum_init` after we've called the external function.

Basically what's happening in all these places is we're passing a memory location to the windows API, which is supposed to write something to that location; so the memory is not safe to access until after that write has happened (and I imagine all of these winapi functions return a flag indicating if there is an error or not, and if there's an error we should also assume the memory is uninitialized...)

Reply:
Yeah, so the current refactoring initializes the memory to safe values and assumes init, which I guess is only _slightly_ safer than the previous use of `mem::uninitialized` - I was just treading the path of least resistance there. The problem then becomes whether passing `MaybeUninit` to these external functions is the same as passing the pointers themselves, or whether we need to do some black magic there. 🤔I wasn't able to tell from the docs how MaybeUninit translates to C.

Reply:
ah okay, I guess the main question I'd have is whether `zeros` is valid data for all the types used here? If it is then that works for me!

Reply:
and yes, passing a pointer to the `MaybeUninit` is the same as passing a pointer to the value; all `MaybeUninit` does is use the type system to force us to be explicit about initialization on the Rust side.

Reply:
This is fancier than needed for RECT, you can just do `mem::zeroed` instead of `mem::uninitialized`. It's a tiny optimization not to zero, but when I wrote the code I was doing things "the C way."

Reply:
Okay, good to know. Now my only concern is with the implications of `GetClientRect` returning false, is that something we want to log and/or return an error?

Reply:
Yes. This is extremely unlikely, but we want to program defensively in any case.

Reply:
Thanks for the feedback, friends! It's updated now. 😄